### PR TITLE
Add UnsafeDeserialization sinks

### PR DESCRIPTION
**Pull request by haby0**
In the study of Java deserialization vulnerabilities, some sinks were recorded. This pr adds these sinks to the CodeQL query.

Mainly known: `JYaml.load`, `JsonReader.jsonToJava`, `YamlReader.read`, `HessianInput.readObject`, `Hessian2Input.readObject`, `Unmarshaller.unmarshal`, `BurlapInput.readObject`. There are also some sink points not mentioned: `YamlConfig.load`,`JsonReader.readObject`.

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/Castor.qll [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * The class `org.exolab.castor.xml.Unmarshaller`.
+ */
+class Unmarshaller extends RefType {
+  Unmarshaller() { this.hasQualifiedName("org.exolab.castor.xml", "Unmarshaller") }
+}
+
+/**
+ * A Unmarshaller unmarshal method. This is either `Unmarshaller.unmarshal`.
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/frameworks/Castor.qll:15]: > This is either `Unmarshaller.unmarshal`
It looks like a part of the sentence is missing "This is either ... or"

File: java/ql/src/semmle/code/java/frameworks/Hessian.qll [hunk 2]
```diff
@@ -22,3 +22,30 @@ class UnSafeHessianInputReadObjectMethod extends Method {
...
+
+/**
+ * The class `com.caucho.burlap.io.BurlapInput`.
+ */
+class BurlapInput extends RefType {
+  BurlapInput() { this.hasQualifiedName("com.caucho.burlap.io", "BurlapInput") }
+}
+
+/**
+ * A BurlapInput readObject method. This is either `BurlapInput.readObject`.
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/frameworks/Hessian.qll:34]: Same here.

File: java/ql/src/semmle/code/java/frameworks/Hessian.qll [hunk 3]
```diff
@@ -22,3 +22,30 @@ class UnSafeHessianInputReadObjectMethod extends Method {
...
+ */
+class BurlapInputReadObjectMethod extends Method {
+  BurlapInputReadObjectMethod() {
+    this.getDeclaringType() instanceof BurlapInput and
+    this.getName() = "readObject"
+  }
+}
+
+/**
+ * A BurlapInput init method. This is either `BurlapInput.init`.
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/frameworks/Hessian.qll:44]: Same here.

**Comment by haby0** [comment]
@intrigus-lgtm I will create a new pr.

**Pull request closed without merging**